frontend/src/components/ClientInstructionCard.vue

```diff
@@ -0,0 +1,125 @@
+    <v-card-text v-else-if="!state.workerStatus.connected && !props.internalHFAccessToken">
+      <div class="mb-2">
+        <v-btn small tile color="primaryLight" class="primaryLightBtn" @click="generateHFToken">Generate</v-btn>
+      </div>
+      <v-row>
+        <v-col>
+          Please enter your Hugging Face access token in Giskard Settings page to generate the instructions.
+        </v-col>
+      </v-row>
+    </v-card-text>
+    <v-card-text v-else>
+      <HuggingFaceTokenCard @submit="fetchAndSaveHFSpacesTokenWithAccessToken"/>
+    </v-card-text>
+  </v-card>
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted } from "vue";
+import { useMainStore } from "@/stores/main";
+import { TYPE } from "vue-toastification";
+
+import CodeSnippet from "@/components/CodeSnippet.vue";
+import LoadingFullscreen from "./LoadingFullscreen.vue";
+import HuggingFaceTokenCard from "./HuggingFaceTokenCard.vue";
+
+import { saveLocalHFToken, getLocalHFToken } from "@/utils";
+import { attemptFetchHFSpacesToken } from "@/hf-utils";
+
+import { apiURL } from "@/env";
+import { state } from "@/socket";
+
+import { JWTToken } from "@/generated-sources";
+import { api } from "@/api";
+
+const mainStore = useMainStore();
+
+interface Props {
+  internalHFAccessToken: boolean,
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  internalHFAccessToken: false,
+});
+
+function generateGiskardClientInstruction(hf_token=null) {
+  let snippet = `from giskard import GiskardClient
+
+url = "${apiURL}"
+token = "${apiAccessToken.value?.id_token}"
+`;
+  if (hf_token) {
+    snippet += `hf_token = "${hf_token}"
+`;
+  }
+  snippet += `
+# Create a giskard client to communicate with Giskard
+client = GiskardClient(url, token`;
+  if (hf_token) {
+    snippet += `, hf_token`;
+  }
+  snippet += `);`;
+  return snippet;
+};
+
+// Giskard API token
+const apiAccessToken = ref<JWTToken | null>(null);
+async function generateToken() {
+  try {
+    apiAccessToken.value = await api.getApiAccessToken();
+  } catch (error) {
+    mainStore.addNotification({content: 'Could not reach server', color: TYPE.ERROR});
+  }
+}
+
+// Hugging Face Spaces token (Giskard Space token)
+const giskardClientTemplate = ref<string | null>(null);
+const needFetchWithHFAccessToken = ref<boolean | null>(null);
+
+async function fetchAndSaveHFSpacesTokenWithAccessToken(accessToken: string) {
+  if (mainStore.appSettings!.isRunningOnHfSpaces) {
+    saveLocalHFToken(accessToken);
+    await attemptFetchHFSpacesToken((token) => {
+      giskardClientTemplate.value = generateGiskardClientInstruction(token);
+      needFetchWithHFAccessToken.value = false;
+    }, () => {
+      needFetchWithHFAccessToken.value = true;
+      mainStore.addNotification({content: 'Invalid Hugging Face access token', color: TYPE.ERROR});
+    });
+  }
+}
+
+async function generateHFToken() {
+  if (mainStore.appSettings?.isRunningOnHfSpaces) {
+    attemptFetchHFSpacesToken((token) => {
```

Comment:
```suggestion
    await attemptFetchHFSpacesToken((token) => {
```

Reply:
There is an initial loading screen before loading, so maybe we don't have to await here?
I just remove the async in onMounted callback

Reply:
@Inokinoki, TBH I don't really undestand the goal of `attemptFetchHFSpacesToken` compared to `fetchHFSpacesToken`

IMO hf-utils.ts should become a store that would:
- contain the last fetched token
- async method to retrieve this token that would 
a) do an API call if a previous token is null or expired
b) return a cached token if it's requested after it was already fetched

Like this we won't have to re-call HF api if we still have a valid token

Reply:
Done